src/Payments.sol

```diff
@@ -1754,6 +1783,47 @@ contract Payments is ReentrancyGuard {
 
         return (fundedUntilEpoch, currentFunds, availableFunds, currentLockupRate);
     }
+
+    /**
+     * @notice Burn FIL to buy the network fees
+     * @param token Which kind of fees to buy
+     * @param recipient Receives the purchased fees
+     * @param requested Exact amount of fees transferred
+     */
+    function burnFILForFees(IERC20 token, address recipient, uint256 requested) external payable {
```

Comment:
It's very hard to use this method to reliably buy a token for a specific price:

1. It'll fail if `requested` tokens aren't available even if there are SOME tokens available. E.g., if many people are racing to participate in the auction at the same time. Ideally, I'd be able to buy (a) at most X tokens (b) at at most Y price.
2. If the price goes down and I send too much FIL, that FIL will get burnt anyways. Really, I'm _guaranteed_ to lose some FIL because it'll always take some time for my message to be included. Ideally, this method would pull the exact amount FIL required from my account within the payment system (this would also ensure that there's exactly ONE way to move funds into and out of the payment system).